### PR TITLE
[docs] Add script to auto generate docs and create pull request while releasing 

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>[docs] Add script to auto generate doc and create pull request while releasing </changelog>`.

### Summary of changes
This pr add the script to auto generate docs and then create pr for review.
It will also create a pr to doc repo to update sdk version number.

### CODE REVIEW

File: scripts/update-android-docs.sh [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+DOKKA_OUTPUT_DIR="./build/dokka/htmlCollector"
+
+readonly ANDROID_DOCS_DIRECTORY="android-docs-repo"
+readonly CONSTANTS_FILE="./src/constants.json"
+readonly MAP_VERSION_NUMBERS_FILE="./src/data/map-version-numbers.json"
+readonly BRANCH_WITH_DOCUMENTATION="publisher-production"
+readonly ANDROID_DOCS_REPO="git@github.com:mapbox/android-docs.git"
+
+MAPS_SDK_VERSION=
+REVIEWERS=
```
Comment (kiryldz) [inline scripts/update-android-docs.sh:24]: Why are those empty?

Comment (pengdev) [reply scripts/update-android-docs.sh:24]: It's initialised below from the command line inputs.

**Comment by pengdev** [comment]
@Chaoba Great work, could you test the CI task as well, to make sure we have all the credentials needed configured in the CI.

File: scripts/update-android-docs.sh (hunk 1 above)
Comment (kiryldz) [reply scripts/update-android-docs.sh:24]: 🙇

**Review by kiryldz — APPROVED** [review]

File: scripts/update-android-docs.sh (hunk 1 above)
Comment (Chaoba) [reply scripts/update-android-docs.sh:24]: They will be used to save the params

**Review by kiryldz — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -295,7 +305,7 @@ jobs:
 
   run-lint:
     docker:
-      - image: mbgl/android-ndk-r21b:latest
+      - image: mbgl/android-ndk-r21:latest
```
Comment (pengdev) [inline .circleci/config.yml:308]: why do we use r21? I think we should use `21e`, which is the latest LTS and align across different teams.

**Review by pengdev — APPROVED** [review]
Great work 🚀

**Pull request merged**